pkgs/by-name/ro/roslyn-ls/package.nix

```diff
@@ -0,0 +1,76 @@
+{ lib, fetchFromGitHub, buildDotnetModule, dotnetCorePackages, stdenvNoCC, testers, roslyn-ls }:
+let
+  pname = "roslyn-ls";
+  # see https://github.com/dotnet/roslyn/blob/main/eng/targets/TargetFrameworks.props
+  dotnet-sdk = with dotnetCorePackages; combinePackages [ sdk_7_0 sdk_8_0 ];
+  # need sdk on runtime as well
+  dotnet-runtime = dotnetCorePackages.sdk_8_0;
+
+  project = "Microsoft.CodeAnalysis.LanguageServer";
+in
+buildDotnetModule rec {
+  inherit pname dotnet-sdk dotnet-runtime;
+
+  vsVersion = "2.16.24";
+  src = fetchFromGitHub {
+    owner = "dotnet";
+    repo = "roslyn";
+    rev = "VSCode-CSharp-${vsVersion}";
+    hash = "sha256-pTZ8cEqDupfZn6SfVd38mJnDEABvXW1S3xaRd0LryoY=";
+  };
+
+  # versioned independently from vscode-csharp
+  # "roslyn" in here:
+  # https://github.com/dotnet/vscode-csharp/blob/main/package.json
+  version = "4.10.0-1.24069.13";
```

Comment:
Does this need to match some version present in the source code?

Reply:
The prefix of this version should match the one defined in [roslyn repo here](https://github.com/dotnet/roslyn/blob/2fe96bca1092f880e91eea6eb17ea3487d89309a/eng/Versions.props#L8). I don't think the other version parts are present in the roslyn repo though.

I choose to follow the more specific version from vscode-csharp repo because those are the official releases and those are the versions that other users (non-nix) will see when downloading via nuget for example [(like here)](https://github.com/jmederosalvarado/roslyn.nvim). Additionally, we are bound to the tags from vscode-csharp either way here.

We could go with just the version from roslyn repo, but the problem is that there may be bugs and fixes without bumping `4.10.0-1` on the roslyn repo which would result in a need to update `4.10.0-1` to... `4.10.0-1` here so wanted to avoid that.